unit_tests/system/file_path_test.cpp

```diff
@@ -17,6 +17,116 @@
 
 using namespace std;
 
+
+TEST(BoostFileSystem, base_methods) {
+	// current_path
+	std::cout << "Current path: " << boost::filesystem::current_path() << std::endl;
+
+	// simple absolute path
+	{
+		stringstream dir;
+		dir << boost::filesystem::current_path().string() << DIR_DELIMITER << "x";
+		if ( !boost::filesystem::is_directory(dir.str()) ) {
+			boost::filesystem::create_directories(dir.str());
+			EXPECT_TRUE( boost::filesystem::is_directory(dir.str()) );
+			boost::filesystem::remove(dir.str());
+			EXPECT_FALSE( boost::filesystem::is_directory(dir.str()) );
+		}
+	}
+
+	// absolute path contains actual dir ('.'), format is current_path() + "/./x"
+	{
+		stringstream dir;
```

Comment:
Either this is the test of create_directories or the test of path joining, which should not be done using redir operators.
Just compare resulting path to a string, or if you are not sure if this is portable implement a single checking method. 
It should be clear what is the reference value or correct behavior of the test. So do not mix tow tests together.